app/launcher/Main.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Main where
+
+import CLI
+    ( Decodable (decode), Encodable (encode), Network, Port (..), getArg )
+import Control.Concurrent
+    ( forkIO )
+import Control.Monad
+    ( forM, forM_, when )
+import GHC.IO.Handle
+    ( Handle )
+import Prelude
+import System.Console.Docopt
+    ( Docopt, docopt, exitWithUsage, isPresent, longOption, parseArgsOrExit )
+import System.Environment
+    ( getArgs )
+import System.Exit
+    ( exitWith )
+import System.Process
+    ( CreateProcess
+    , ProcessHandle
+    , cleanupProcess
+    , createProcess
+    , interruptProcessGroupOf
+    , proc
+    , waitForProcess
+    )
+
+
+-- | Command-Line Interface specification. See http://docopt.org/
+cli :: Docopt
+cli = [docopt|
+cardano-wallet-launcher
+
+Start the cardano wallet along with its API and underlying node.
+
+Requires cardano-http-bridge. To install, follow instructions at
+https://github.com/input-output-hk/cardano-http-bridge, and run
+  cargo install --path .
+in the directory.
+
+
+Usage:
+  cardano-wallet-launcher [options]
+  cardano-wallet-launcher --help
+
+Options:
+  --wallet-server-port <PORT>  port used for serving the wallet API [default: 8090]
+  --node-port <PORT>           port used for node-wallet communication [default: 8080]
+  --network <NETWORK>          mainnet or testnet [default: mainnet]
+|]
+
+
+type Process = (Maybe Handle, Maybe Handle, Maybe Handle, ProcessHandle)
+
+main :: IO ()
+main = do
+    args <- parseArgsOrExit cli =<< getArgs
+    when (args `isPresent` (longOption "help")) $ exitWithUsage cli
+    let getArg' = getArg args cli
+
+    nodePort <- getArg' (longOption "node-port") decode
+    walletPort <- getArg' (longOption "wallet-server-port") decode
+    network <- getArg' (longOption "network") decode
+
+    putStrLn $
+        "Starting wallet on port " ++ (encode walletPort) ++
+        ",\n         connecting to node on port " ++ (encode nodePort)
+
+    processes <- forM
+        [ nodeHttpBridgeOn nodePort network
+        , walletOn walletPort nodePort
+        ] createProcess
```

Comment:
Side-note on style / readability, `mapM` is usually a better choice when your function is short and your iteree long. Basically, `forM = flip mapM`, so they are the same thing, but IMO, they read better depending on their arguments.

```hs
-- Iteree is short
forM xs $ \x -> do ...  -- Feels right
mapM (\x -> do ....) xs  -- Feels akward

-- Function is short
forM [many, different, things, because, there, is, no, intermediate, var] fn -- Feels akward
mapM fn [many, different, things, because, there, is no, intermediate, var] -- Feels right
```

This is really neat-picking though, I don't mind much as it is :+1:

Reply:
👍 Thanks! Agreed.